spec.md

```diff
@@ -144,18 +144,57 @@ circuit MyTop...
 
 ## Circuits
 
-All FIRRTL circuits consist of a list of modules, each representing a hardware
-block that can be instantiated. The circuit must specify the name of the
-top-level module.
+A FIRRTL circuit is a named collection of FIRRTL modules.  Each module is a
+hardware "unit" that has ports, registers, wires, and may instantiate other
+modules (see: [@sec:modules]).  (This is the same concept as a Verilog
+`module`{.verilog}.)  A public module is usable outside of the current circuit.
+
+Consider the following circuit named `Foo`.  This contains two modules, `Bar`
+and `Baz`.  Module `Baz` is marked public.
 
 ``` firrtl
-circuit MyTop :
-  module MyTop :
-    ; ...
-  module MyModule :
-    ; ...
+circuit Foo :
+  module Bar :
+    input a: UInt<1>
+    output b: UInt<1>
+
+    connect b, a
+
+  public module Baz :
+    input a: UInt<1>
+    output b: UInt<1>
+
+    inst bar of Bar
+    connect bar.a, a
+    connect b, bar.b
+```
+
+A circuit may or may not contain a module with the same name as the circuit.  A
+circuit may contain public modules that instantiate no common instances.  Two
+public modules may share common instances.
+
+Circuit `Foo`, below, shows an example of a more complex circuit.  This circuit
+contains three public modules, `Foo`, `Bar`, and `Baz`, and one non-public
+module, `Qux`.  Module `Foo` has no common instances with `Bar` or `Baz`.
+`Bar` and `Baz` both instantiate `Qux`:
+
+``` firrtl
+circuit Foo :
+  public module Foo :
+
+  public module Bar :
+    inst qux of Qux
+
+  public module Baz :
+    inst qux of Qux
+
+  module Qux :
 ```
 
+A circuit that contains no public modules is trivially equivalent to a circuit
+that contains no modules.  It is not enforced that a circuit has at least one
+public module, but it is expected that it does.
```

Comment:
Is the empty circuit legal in firrtl currently?

I think I'm find with the empty circuit being legal.